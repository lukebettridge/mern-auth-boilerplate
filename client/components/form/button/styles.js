import styled, { css } from "styled-components";

const Button = styled.button`
	background-color: #f8f8f8;
	border: 1px solid #eeeeee;
	cursor: pointer;
	font-size: 13pt;
	font-weight: 600;
	max-width: 350px;
	padding: 8px 20px;
	width: 100%;

	${props =>
		props.as === "a" &&
		css`
			background-color: transparent;
			color: #596f86;
			display: block;
			font-weight: 400;
			max-width: none;
			text-decoration: none;
			&:hover {
				text-decoration: underline;
			}

			${props.secondary &&
				css`
					color: #aaaaaa;
					font-size: 11pt;
					font-weight: 300;
				`}
		`}

	${props =>
		props.mt === "s" &&
		css`
			margin-top: 16px;
		`}

	${props =>
		props.mb === "s" &&
		css`
			margin-bottom: 16px;
		`}
`;

module.exports = { Button };
