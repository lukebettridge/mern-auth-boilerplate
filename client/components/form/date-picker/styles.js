import styled, { css } from "styled-components";

const DatePickerContainer = styled.div`
	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}

	${props =>
		(props.value || props.focused) &&
		css`
			& + label {
				font-size: 95%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}
`;

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Container = styled.div`
	position: relative;

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}

	.SingleDatePicker,
	.SingleDatePickerInput,
	.DateInput {
		display: block;
		width: 100%;
	}

	.DateInput_input {
		background-color: transparent;
		border-bottom: 1px solid #e9ecef;
		font-size: 11pt;
		font-weight: 300;
		letter-spacing: 1px;
		margin-bottom: 8px;
		padding: 12px 10px 5px;
		transition: border-color 100ms;
		width: 100%;

		&:hover {
			border-color: #b3b3b3;
		}

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}
	}
`;

const Label = styled.label`
	color: #999999;
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Container, DatePickerContainer, Error, Label };
