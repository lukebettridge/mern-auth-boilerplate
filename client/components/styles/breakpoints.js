module.exports = {
	xxs: "400px",
	xs: "450px",
	s: "590px",
	m: "768px",
	l: "900px",
	xl: "950px",
	xxl: "1200px"
};
