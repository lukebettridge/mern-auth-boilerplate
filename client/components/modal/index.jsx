import React, { useEffect } from "react";
import PropTypes from "prop-types";

import * as S from "./styles";

const Modal = props => {
	useEffect(() => {
		S.Modal.setAppElement("#app");
	}, []);

	return (
		<S.Modal
			isOpen={props.isOpen}
			onRequestClose={props.close}
			sideModal={props.sideModal}
		>
			<S.Header>
				<S.Close onClick={props.close} />
				{props.title && <S.Heading>{props.title}</S.Heading>}
			</S.Header>
			{props.children}
		</S.Modal>
	);
};

Modal.propTypes = {
	children: PropTypes.any,
	close: PropTypes.func,
	isOpen: PropTypes.bool.isRequired,
	sideModal: PropTypes.bool,
	title: PropTypes.string
};

export default Modal;
