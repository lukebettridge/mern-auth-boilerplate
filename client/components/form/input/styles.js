import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Input = styled.input`
	background-color: transparent;
	border: 0;
	border-bottom: 1px solid #e9ecef;
	box-shadow: none;
	font-size: 11pt;
	font-weight: 300;
	margin-bottom: 8px;
	outline: none;
	padding: 12px 10px;
	width: 100%;

	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			border-color: #db1802;

			& + label {
				color: #db1802;
			}
		`}
`;

const InputContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 20px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Input, InputContainer, Label };
