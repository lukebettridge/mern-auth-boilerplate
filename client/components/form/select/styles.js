import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";
import { default as ReactSelect } from "react-select";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Select = styled(ReactSelect)`
	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		props.value.length > 0 &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}
        
	.react-select__control {
		appearance: none;
		background-color: transparent;
		border: 0;
		border-bottom: 1px solid #e9ecef;
		border-radius: 0;
		box-shadow: none;
		font-size: 12pt;
		font-weight: 300;
		height: 42px;
		margin-bottom: 8px;
		outline: none;
		width: 100%;

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}

		.react-select__input {
			opacity: 0;
		}
	}
`;

const SelectContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Select, SelectContainer, Label };
