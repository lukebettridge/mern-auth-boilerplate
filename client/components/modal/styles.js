import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import ReactModal from "react-modal";

const Body = styled.div`
	@media (min-width: ${breakpoints.m}) {
		padding: 10px 20px;
	}

	${props =>
		!props.sideModal &&
		css`
			text-align: center;
		`}
`;

const Close = styled.a`
	height: 32px;
	opacity: 0.3;
	position: relative;
	width: 32px;

	&:hover {
		cursor: pointer;
		opacity: 1;
	}

	&:before,
	&:after {
		background-color: #333;
		content: " ";
		height: 33px;
		left: 15px;
		position: absolute;
		width: 2px;
	}

	&:before {
		transform: rotate(45deg);
	}

	&:after {
		transform: rotate(-45deg);
	}
`;

const Header = styled.div`
	display: flex;
	flex-direction: row-reverse;
	justify-content: space-between;
	margin-bottom: 30px;

	@media (min-width: ${breakpoints.m}) {
		padding-left: 20px;
	}
`;

const Heading = styled.h2`
	font-weight: 700;
	margin: 0;
`;

const Modal = styled(ReactModal)`
	background-color: #ffffff;
	border: 1px solid #e9ecef;
	bottom: auto;
	left: 50%;
	max-width: 650px;
	outline: none;
	padding: 20px 20px 62px;
	position: absolute;
	right: auto;
	top: 43%;
	width: calc(100% - 20px);

	${props =>
		props.sideModal
			? css`
					border: 0;
					border-left: 1px solid #e9ecef;
					bottom: 0;
					left: auto;
					max-width: 720px;
					overflow-y: auto;
					padding: 40px 20px;
					right: -100%;
					top: 0;
					transition: right 0.5s;
					width: 100%;
					z-index: 2;

					&.ReactModal__Content--after-open {
						right: 0;
					}

					@media (min-width: ${breakpoints.m}) {
						padding: 60px;
					}
			  `
			: css`
					border-radius: 2px;
					padding-bottom: 62px;
					transform: translate(-50%, -50%);
			  `}
`;

module.exports = { Body, Close, Header, Heading, Modal };
