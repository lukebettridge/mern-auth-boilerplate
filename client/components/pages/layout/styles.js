import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import { Link as RouterLink } from "react-router-dom";

const anchorStyles = `
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}
`;

const subAnchorStyles = `
	${anchorStyles}
	font-size: 11pt;
	padding-left: 15px;
`;

const Anchor = styled.a`
	${anchorStyles}
	${props =>
		props.active &&
		css`
			opacity: 1;
		`}
`;

const Container = styled.div`
	display: flex;
	height: 100%;
`;

const Link = styled(RouterLink)`
	${anchorStyles}
`;

const List = styled.ul`
	margin: 0;
`;

const ListItem = styled.li`
	margin-bottom: 40px;
`;

const Slider = styled.div`
	align-items: center;
	background-color: #fbfbfc;
	border-left: 1px solid #e9ecef;
	border-bottom: 1px solid #e9ecef;
	color: #294661;
	display: flex;
	height: 50px;
	justify-content: center;
	position: absolute;
	right: 0;
	top: 0;
	width: 50px;
	z-index: 1;
	@media (min-width: ${breakpoints.m}) {
		display: none;
	}
`;

const SubAnchor = styled.a`
	${subAnchorStyles}
`;

const SubLink = styled(RouterLink)`
	${subAnchorStyles}
`;

const SubList = styled.ul`
	margin-top: 15px;
	max-height: 1000px;
	overflow: hidden;
	transition: all 0.3s;

	${props =>
		!props.active &&
		css`
			margin: 0;
			max-height: 0;
		`}
`;

const SubListItem = styled.li`
	margin-bottom: 8px;
`;

const Navigation = styled.div`
	background-color: #fbfbfc;
	border-right: 1px solid #e9ecef;
	display: none;
	height: 100%;
	max-width: 200px;
	min-width: 200px;
	padding-left: 15px;
	padding-right: 10px;
	padding-top: 48px;
	width: 200px;
	z-index: 1;

	${props =>
		props.active &&
		css`
			display: table-cell;
			position: absolute;
		`}

	@media (min-width: ${breakpoints.m}) {
		display: table-cell;
		height: unset;
		position: unset;
	}
`;

module.exports = {
	Anchor,
	Container,
	Link,
	List,
	ListItem,
	Navigation,
	Slider,
	SubAnchor,
	SubLink,
	SubList,
	SubListItem
};
