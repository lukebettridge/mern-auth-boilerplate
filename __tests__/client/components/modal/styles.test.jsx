import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";

import breakpoints from "components/styles/breakpoints";
import { Body, Close, Header, Heading, Modal } from "components/modal/styles";

jest.mock("react-modal", () => "div");

describe("Modal styles", () => {
	it("snapshot renders", () => {
		const subject = mount(
			<Modal>
				<Header>
					<Close />
					<Heading />
				</Header>
				<Body />
			</Modal>
		);

		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("Body: omit align center for side modal", () => {
		const subject = mount(<Body sideModal />);

		expect(subject).not.toHaveStyleRule("text-align", "center");
	});

	it("Modal: side modal styles applied", () => {
		const subject = mount(<Modal sideModal />);

		expect(subject).toHaveStyleRule("border", "0");
		expect(subject).toHaveStyleRule("border-left", "1px solid #e9ecef");
		expect(subject).toHaveStyleRule("bottom", "0");
		expect(subject).toHaveStyleRule("left", "auto");
		expect(subject).toHaveStyleRule("max-width", "720px");
		expect(subject).toHaveStyleRule("padding", "40px 20px");
		expect(subject).toHaveStyleRule("right", "-100%");
		expect(subject).toHaveStyleRule("top", "0");
		expect(subject).toHaveStyleRule("transition", "right 0.5s");
		expect(subject).toHaveStyleRule("width", "100%");
		expect(subject).toHaveStyleRule("z-index", "2");
		expect(subject).toHaveStyleRule("right", "0", {
			modifier: "&.ReactModal__Content--after-open"
		});
		expect(subject).toHaveStyleRule("padding", "60px", {
			media: `(min-width: ${breakpoints.m})`
		});
	});

	it("Modal: non side modal styles applied", () => {
		const subject = mount(<Modal />);

		expect(subject).toHaveStyleRule("border-radius", "2px");
		expect(subject).toHaveStyleRule("padding-bottom", "62px");
		expect(subject).toHaveStyleRule("transform", "translate(-50%,-50%)");
	});
});
