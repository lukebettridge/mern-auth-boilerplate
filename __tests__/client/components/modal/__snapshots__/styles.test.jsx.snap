// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal styles snapshot renders 1`] = `
<styles__Modal>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "styles__Modal-sc-6vcceo-4",
          "isStatic": false,
          "lastClassName": "cFTjBz",
          "rules": Array [
            "background-color:#ffffff;border:1px solid #e9ecef;bottom:auto;left:50%;max-width:650px;outline:none;padding:20px 20px 62px;position:absolute;right:auto;top:43%;width:calc(100% - 20px);",
            [Function],
          ],
        },
        "displayName": "styles__Modal",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "styles__Modal-sc-6vcceo-4",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="cFTjBz"
    >
      <styles__Header>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "styles__Header-sc-6vcceo-2",
                "isStatic": false,
                "lastClassName": "fZzAoB",
                "rules": Array [
                  "display:flex;flex-direction:row-reverse;justify-content:space-between;margin-bottom:30px;@media (min-width:",
                  "768px",
                  "){padding-left:20px;}",
                ],
              },
              "displayName": "styles__Header",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "styles__Header-sc-6vcceo-2",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="fZzAoB"
          >
            <styles__Close>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "styles__Close-sc-6vcceo-1",
                      "isStatic": false,
                      "lastClassName": "dbJOmm",
                      "rules": Array [
                        "height:32px;opacity:0.3;position:relative;width:32px;&:hover{cursor:pointer;opacity:1;}&:before,&:after{background-color:#333;content:\\" \\";height:33px;left:15px;position:absolute;width:2px;}&:before{transform:rotate(45deg);}&:after{transform:rotate(-45deg);}",
                      ],
                    },
                    "displayName": "styles__Close",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "styles__Close-sc-6vcceo-1",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <a
                  className="dbJOmm"
                />
              </StyledComponent>
            </styles__Close>
            <styles__Heading>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "styles__Heading-sc-6vcceo-3",
                      "isStatic": false,
                      "lastClassName": "cTnCmE",
                      "rules": Array [
                        "font-weight:700;margin:0;",
                      ],
                    },
                    "displayName": "styles__Heading",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "styles__Heading-sc-6vcceo-3",
                    "target": "h2",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <h2
                  className="cTnCmE"
                />
              </StyledComponent>
            </styles__Heading>
          </div>
        </StyledComponent>
      </styles__Header>
      <styles__Body>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "styles__Body-sc-6vcceo-0",
                "isStatic": false,
                "lastClassName": "dfQguO",
                "rules": Array [
                  "@media (min-width:",
                  "768px",
                  "){padding:10px 20px;}",
                  [Function],
                ],
              },
              "displayName": "styles__Body",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "styles__Body-sc-6vcceo-0",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="dfQguO"
          />
        </StyledComponent>
      </styles__Body>
    </div>
  </StyledComponent>
</styles__Modal>
`;
